src/gadgets/utils.rs

```diff
@@ -80,15 +80,7 @@ where
   CS: ConstraintSystem<<E as Engine>::Base>,
 {
   AllocatedNum::alloc(cs.namespace(|| "allocate scalar as base"), || {
-    let input_bits = input.unwrap_or(E::Scalar::ZERO).clone().to_le_bits();
-    let mut mult = E::Base::ONE;
-    let mut val = E::Base::ZERO;
-    for bit in input_bits {
-      if bit {
-        val += mult;
-      }
-      mult = mult + mult;
-    }
```

Comment:
Could this be a one-liner?

`AllocatedNum::alloc(cs.namespace(|| "allocate scalar as base"), || Ok(scalar_as_base::<E>(input.unwrap_or(E::Scalar::ZERO))))`

Reply:
It is morally a one-liner, you're commenting on the removed code. Unless you're talking specifically about inlining the val variable, which I'll take as a nit?